### PR TITLE
Refactor yobserver

### PR DESCRIPTION
**Pull request by gularsson**

### CODE REVIEW

File: src/hooks/useYObserver.tsx [hunk 1]
```diff
@@ -1,73 +1,109 @@
...
-export function useYObserver<T>(y: Y.Map<unknown> | undefined, key: string): [T, YSetter]
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function useYObserver<T extends Record<string, any>>(y: Y.Map<unknown> | undefined, key?: string): [Record<string, T>, YSetter]
-export function useYObserver<T>(y: Y.Array<unknown> | undefined, key?: string): [T[], YSetter]
-export function useYObserver<T>(y: Y.Map<unknown> | Y.Array<unknown> | undefined, key?: string): [T | Record<string, T> | T[], YSetter] {
-  const [value, setValue] = useState<unknown | undefined>(get(y as Y.Map<unknown>, key))
+  // Get wanted Y.Map by path provided
+  const map = yMapValueByPath.get(yPlanning, path)
 
+  // Observe whole yPlanning to detect changes at top
```
Comment (dannelundqvist) [inline src/hooks/useYObserver.tsx:36]: Are the two comments misleading? It's not just planning we're tracking?

**Review by dannelundqvist — CHANGES_REQUESTED** [review]
A thorough and solid refactor. I'll probably have more to say about it after using the code a bit more. Found two slightly misleading comments though. Other than that - great job!

File: src/hooks/useYObserver.tsx (hunk 1 above)
Comment (gularsson) [reply src/hooks/useYObserver.tsx:36]: I would say it's accurate, it's just the planning/yPlanning we're observing. We're getting the planning ymap and attaching the observeDeep on that:
```
const yPlanning = document?.getMap('planning')
...
    yPlanning?.observeDeep((events) => {
```
Changes to `open-documents` and `content` shouldn't be observed afaik.

It's possible that the second observer (on the `map`) is superfluous. We should be able to catch all the changes on the observeDeep.

Comment (gularsson) [reply src/hooks/useYObserver.tsx]: Now i see, I'll fix so we also support the `article` Y.Map (and others)

**Review by dannelundqvist — APPROVED** [review]
Good solution, seems really easy to use now!

**Pull request merged**